composer.json

```diff
@@ -7,13 +7,15 @@
     "type": "library",
     "require": {
         "php" : "^7.1 || ^8.0",
-        "php-http/client-common": "^2.0",
+        "psr/http-message-implementation": "*",
+        "psr/http-factory-implementation": "*",
+        "psr/http-client-implementation": "*",
         "php-http/discovery": "^1.7",
-        "nyholm/psr7": "^1.0",
         "ext-json": "*"
     },
     "require-dev": {
         "php-http/guzzle6-adapter": "^2.0",
+        "guzzlehttp/psr7": "^2.0",
```

Comment:
i suggest to use the guzzle psr 7 implementation when using the guzzle client. but we can also switch back to nyholm, it should not break anything either way.

Reply:
ok, with guzzle 6, we would need psr7 version 1 which does not include the factories. i suggest going with guzzle 7 for the tests. that is also compatible with php 8.